python/lsst/ap/association/diaPipe.py

```diff
@@ -0,0 +1,229 @@
+#
+# LSST Data Management System
+# Copyright 2008-2016 AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+
+"""PipelineTask for associating DiaSources with previous DiaObjects.
+
+Additionally performs forced photometry on the calibrated and difference
+images at the updated locations of DiaObjects.
+
+Currently loads directly from the Apdb rather than pre-loading.
+"""
+
+import os
+
+import lsst.dax.apdb as daxApdb
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+import lsst.pipe.base.connectionTypes as connTypes
+from lsst.utils import getPackageDir
+
+from lsst.ap.association import (
+    AssociationTask,
+    DiaForcedSourceTask,
+    LoadDiaCatalogsTask,
+    MapDiaSourceTask,
+    make_dia_object_schema,
+    make_dia_source_schema)
+
+__all__ = ("DiaPipelineConfig",
+           "DiaPipelineTask",
+           "DiaPipelineConnections")
+
+
+class DiaPipelineConnections(pipeBase.PipelineTaskConnections,
+                             dimensions=("instrument", "visit", "detector"),
+                             defaultTemplates={}):
+    """Butler connections for DiaPipelineTask.
+    """
+    diaSourceSchema = connTypes.InitInput(
+        doc="Schema of the DiaSource catalog produced during image "
+            "differencing",
+        name="deepDiff_diaSrc_schema",
+        storageClass="SourceCatalog",
+        multiple=True
+    )
+    diaSourceCat = connTypes.Input(
+        doc="Catalog of DiaSources produced during image differencing.",
+        name="deepDiff_diaSrc",
+        storageClass="SourceCatalog",
+        dimensions=("instrument", "visit", "detector"),
+    )
+    diffIm = connTypes.Input(
+        doc="Difference image on which the DiaSources were detected.",
+        name="deepDiff_differenceExp",
```

Comment:
Please use a template instead of hardcoding the "deep". I assume the Gen 3 differencing task, like the old one, will still have multiple types of diffs.
```suggestion
        name="{coaddName}Diff_differenceExp",
```